### PR TITLE
Completed Issue (fixed)

### PR DESCRIPTION
**Pull request by maxwaldman8**

### CODE REVIEW

File: AstraEngine.Core/Component.cs [hunk 1]
```diff
@@ -13,6 +13,8 @@ public abstract class Component
     /// </summary>
     [AllowNull]
     public Entity Entity { get; internal set; }
+    /// <summary>Whether or not this <see cref="Component"/> is active</summary>
+    public bool Active { get; set; } = true;
```
Comment (jcollard) [inline AstraEngine.Core/Component.cs:17]: Boolean naming convention says to use affirmative phrases e.g. `IsActive`. See: https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members

File: AstraEngine.Core/Component.cs [hunk 2]
```diff
@@ -30,4 +32,12 @@ public virtual void Tick(double delta)
     {
 
     }
+
+    /// <summary>
+    /// Executed when this component exits the game
+    /// </summary>
+    public virtual void Exit()
```
Comment (jcollard) [inline AstraEngine.Core/Component.cs:39]: Can you provide an example use case for this? When would exit be called? Who is the caller?

File: AstraEngine.Core/Entity.cs [hunk 3]
```diff
@@ -8,6 +8,10 @@ namespace AstraEngine.Core;
 /// </summary>
 public class Entity
 {
+    /// <summary>Name for the entity</summary>
+    public string Name { get; set; } = "";
```
Comment (jcollard) [inline AstraEngine.Core/Entity.cs:12]: Microsoft guidelines suggest using `string.Empty` rather than `""`.  See: https://learn.microsoft.com/en-us/dotnet/api/system.string.empty?view=net-8.0

File: AstraEngine.Core/Entity.cs [hunk 4]
```diff
@@ -167,4 +231,14 @@ public void AddChild(Entity player)
     {
         player.Parent = this;
     }
+
+    /// <summary>
+    /// Removes the specified Entity as a child of this component
+    /// </summary>
+    /// <param name="player"></param>
+    public void RemoveChild(Entity player)
```
Comment (jcollard) [inline AstraEngine.Core/Entity.cs:239]: Suggest renaming `player` to `entity` or `child`.

File: AstraEngine.Core/Entity.cs [hunk 5]
```diff
@@ -167,4 +231,14 @@ public void AddChild(Entity player)
...
         player.Parent = this;
     }
+
+    /// <summary>
+    /// Removes the specified Entity as a child of this component
+    /// </summary>
+    /// <param name="player"></param>
+    public void RemoveChild(Entity player)
+    {
+        player.Parent = null;
```
Comment (jcollard) [inline AstraEngine.Core/Entity.cs:241]: * What happens when the entity is not a child of this Entity?
* What happens if the entity was already a child of another Entity?

**Review by jcollard — CHANGES_REQUESTED** [review]
This PR seems to be adding several features, typically, each one would be its own PR:

* IsActive - This code looks good but you should add unit tests for it
* Introduces an Exit method - You should provide a use case for this as well as update the engine to call it. How do we know to call it? When is it called? and who is the caller? Is this analogous to Unity's OnDestroy()? 
* Name + finder methods - These seem fine. Simple Unit tests / examples should be added
* RemoveChild - Unit tests for this should be added showing use case as well as edge cases.

**Review by jcollard — APPROVED** [review]
These changes look great! Nice effort here.
